ui/src/wizard/LocationListStep.ts

```diff
@@ -82,24 +137,31 @@ export class LocationListStep<T extends ILocationWizardContextInternal> extends
     }
 }
 
-async function getAllLocations(wizardContext: ILocationWizardContextInternal): Promise<types.AzExtLocation[]> {
+function generalizeLocationName(name: string | undefined): string {
+    return (name || '').toLowerCase().replace(/[^a-z0-9]/gi, '');
```

Comment:
with toLowerCase, the 'i' flag is unnecessary.

Reply:
@ravipal I didn't have the `i` here and you commented about it! 😝  https://github.com/microsoft/vscode-azuretools/pull/896#discussion_r626975720. I figured it made it more readable and doesn't hurt so I included it

@wwlorey `generalizeLocationName` is used to test equality of locations because different APIs use slightly different naming. I want "North Central US (Stage)" to be equal to "northcentralusstage", so my main reason was to remove those parentheses. I've never seen any other special characters in the location names